File: src/Test/AckermannFFICheatcode.erl
-module(test_ackermannFFICheatcode@foreign).
-export([runAckermannFFICheatcode/1]).
runAckermannFFICheatcode(M) -> ack(M, 4).
ack(0, N) -> N + 1;
ack(M, 0) -> ack(M - 1, 1);
ack(M, N) -> ack(M - 1, ack(M, N - 1)).

File: src/Test/AckermannFFICheatcode.js
export const runAckermannFFICheatcode = function(ignore) {
  return ack(3, 4);
};

function ack(m, n) {
  if (m === 0) return n + 1;
  if (m > 0 && n === 0) return ack(m - 1, 1);
  return ack(m - 1, ack(m, n - 1));
}
